src/app/pages/system/general-settings/localization-form3/components/ix-input/ix-input.component.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  Component, EventEmitter, forwardRef, Input, Output,
+} from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { UntilDestroy } from '@ngneat/until-destroy';
+
+@UntilDestroy()
+@Component({
+  selector: 'ix-input',
+  styleUrls: ['./ix-input.component.scss'],
+  templateUrl: './ix-input.component.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => IxInput),
+      multi: true,
+    },
+  ],
+})
+export class IxInput implements ControlValueAccessor {
+  @Input() label: string;
+  @Input() placeholder: string;
+  @Input() prefixText: string;
+  @Input() suffixIcon: string;
+  @Output() suffixIconClick: EventEmitter<MouseEvent> = new EventEmitter();
+  @Input() hint: string;
+  @Input() tooltip: string;
+  @Input() required: boolean;
+
+  value: string | number = '';
```

Comment:
Is value really a number?

Reply:
Sometimes, it can be. I've seen instances where value is an id which is an integer. That can be easily casted to a string either way but I was going for a bit more flexibility.